lib/identity_cache.rb

```diff
@@ -131,6 +134,14 @@ def fetch_multi(*keys)
       result
     end
 
+    def with_fetch_returns_relation(value = true)
+      previous_value = self.fetch_returns_relation
+      self.fetch_returns_relation = value
+      yield
+      ensure
```

Comment:
This should be indented only as much as `def`.  The following line also needs less indentation